game/src/main/scala/io/aigar/game/GameThread.scala

```diff
@@ -70,14 +77,23 @@ class GameThread(scoreThread: ScoreThread) extends Runnable
   }
 
   def transferActions: Unit = {
-    while (!actionQueue.isEmpty) {
-      val action = actionQueue.take
+    var actions = new java.util.ArrayList[ActionQueryWithId]()
+    actionQueue.drainTo(actions)
```

Comment:
It's kinda like copying the queue to a list and emptying it. There's too much code here since I'm using java primitive in scala.

I'm doing it like that for 2 reasons. First, I don't think I can cleanly iterate through the queue in scala and it's to not block the processes. While I'm iterating, the main thread can still safely put new thing in it.